rfcs/2021-07-20-8288-csv-enrichment.md

```diff
@@ -189,14 +216,20 @@ asynchronously.
 - What approach should we taken around determining the indexed fields?
 - Can we assume that if the file is encrypted it will use AES?
 - Should `title_row` default to true or false?
+- Does the parameter `case_insensitive` indicate that perhaps does a simple 
+  key/value search is not sufficient.
 
 ## Plan Of Attack
 
-Incremental steps that execute this change. Generally this is in the form of:
+- [ ] Add support for resources. Any resource sections will load the data at 
+      boot time and will prevent Vector from starting up if the resource file
+      cannot be found or is incorrectly formatted.
```

Comment:
This alone seems to be worthy of a separate RFC, I'm not sure what you, @jszwedko and @binarylogic agreed on, but I would advocate for one RFC that specifically focusses on the addition of the "resource" type, which also describes (not in exact details, but exact enough that we _know_ it's doable) how we're going to leverage the resource type for other types of resources (database, external URL), and how we're going to handle encrypted resources.

Once that RFC lands, this RFC can simply reference that one, and expand on it by explaining how the resource is passed to VRL, how you reference your resource, and how the actual function is going to work.

Reply:
I agree, although it feels premature at this point. I'd much rather wait until we introduce another resource type to consider this. Hence why I wasn't onboard with resources to begin with.

I think it's fine to introduce for now, as long as it's CSV specific.

Reply:
> I think it's fine to introduce for now, as long as it's CSV specific.

Just to clarify, that means you're okay with making breaking changes if we find out we need to tweak the layout of the resource configuration if it turns out to be inflexible w.r.t. other resource types? Or do you mean we should settle that in this RFC and make sure we allow forward compatibility with resources such as databases, remote URLs, etc?

Reply:
Sure, to be more specific, I want ot make sure we are not creating artificial restrictions when we decide to introduce another resource type, like a database. We should have complete free to integrate with that resource type in whatever way is most optimal. In regards to the options, I'd like to consider a patter that will allow for other resource types. Let's do our best for now. Just like anything, I'm ok breaking backward compatibility if it's worth it.